src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -357,6 +357,66 @@ private static byte[] UrlEncode(byte[] bytes, int offset, int count)
             return expandedBytes;
         }
 
+        private static unsafe string UrlEncodeInternal(string originalString)
+        {
+            fixed (char* pch = originalString)
+            {
+                int safeCharCount = 0;
+                int spaceCharCount = 0;
+                for (int i = 0; i < originalString.Length; i++)
+                {
+                    char c = pch[i];
+                    if (c == ' ')
+                    {
+                        spaceCharCount++;
```

Comment:
This looks exactly like the loop in code that handles bytes. Make them look same (same variable names, formatting, ...).